pkg/dataServer/dataServer.go

```diff
@@ -81,7 +77,7 @@ func (ds *DataServer) Start(ctx context.Context, exitCh chan int) error {
 		level.Info(ds.logger).Log("msg", "runner signaled it is ready")
 		ds.readyChannel <- true
 		level.Info(ds.logger).Log("msg", "dataServer ready for use")
-		<-ds.exitCh
+		<-done
```

Comment:
Why not use the context here?
```suggestion
		<-ctx.Done()
```

Reply:
Yeah, good idea!